### ISSUE TITLE
Add a slider / carousel as section background to use in parallax

### ISSUE THREAD

**Issue by yeshansachithak** [pojome/elementor#7045]
I'm using Elementor Pro and I'm trying to create a website land page like below website with parallax effect in Elementor. As far as I know, there is no option to add slider to section background. With a single background image I can have parallax effect when scrolling. Tried with Slider Widget and Carousel Widget. But, I was unable to complete it. 

https://helmicksculpture.com/

How do I create a home page like above sample website?

Thanks!

**Comment by shilo-ey** [comment]
I merged this issue with  #2898, which is on the same subject so we could deal with it more efficiently. 
Please feel free to continue the conversation there.

Thanks for your feedback!

**Comment by yeshansachithak** [comment]
[SAMPLE WEBSITE](https://helmicksculpture.com/)

I was trying to implement the same slider with parallax scroll effect like above website using Elementor Pro. As per Elementor support we cannot implement that. 

AE Pro have an option to add slider images. But it didn't work with parallax. Also, AE Pro support and essential-addons confirmed that their plugins are not support for my requirement. 

Then I followed [this](http://snifflevalve.com/elementor-tutorials/elementor-parallax-section-background-image/ ) article to implement parallax. Which was perfect for one background image. 

With using above tutorial. I created below ```bgSlider()``` function to full fill my requirement. Hope this will help someone. 

```php
function jquery_parallax()
{
    ?>
    <script language="JavaScript" type="text/javascript">
        (function ($) {
            'use strict';
            var testMobile;
            var isMobile = {
                Android: function () {
                    return navigator.userAgent.match(/Android/i);
                },
                BlackBerry: function () {
                    return navigator.userAgent.match(/BlackBerry/i);
                },
                iOS: function () {
                    return navigator.userAgent.match(/iPhone|iPad|iPod/i);
                },
                Opera: function () {
                    return navigator.userAgent.match(/Opera Mini/i);
                },
                Windows: function () {
                    return navigator.userAgent.match(/IEMobile/i);
                },
                any: function () {
                    return (isMobile.Android() || isMobile.BlackBerry() || isMobile.iOS() || isMobile.Opera() || isMobile.Windows());
                }
            };

            var parallax = function () {
                testMobile = isMobile.any();
                if (testMobile == null) {
                    $(".parallax").parallax("50%", 0.3);
                }
            };

            //Step 01: Copy below till `End Added by Yesh`
            //Added by Yesh
            var CONSOLE_MODE = false;

            function bgSlider(a) {
                (CONSOLE_MODE) ? console.log('bgSlider :::: called') : "";
                this.imageList = [
                    'http://myweb.com/wp-content/uploads/2019/02/Nidakule-Guney-Aukett-Swanke-2.jpg',
                    'http://myweb.com/wp-content/uploads/2019/02/Tefken-Oz-Levent-Ofis-Aukett-Swanke-3.jpg',
                    'http://myweb.com/wp-content/uploads/2019/02/Lesso-Mall-Aukett-Swanke-2.jpg',
                    'http://myweb.com/wp-content/uploads/2019/02/Lillie-Square_Veretec-5.jpg',
                    'http://myweb.com/wp-content/uploads/2019/02/Radisson-Hotel-Tunisia-Aukett-Swanke-3.jpg'
                ];
                this.widgetID = a;
                this.widgetElm = "";
                this.itemDuration = 5000;
                this.rotateTick = 0;
                this.nxtElem = 0;
                this.unInit = function () {
                    (CONSOLE_MODE) ? console.log('bgSlider :: unInit :: called') : "";
                    clearTimeout(this.rotateTick);
                };
                this.bgPreload = function () {
                    $(this.imageList).each(function () {
                        $('<img/>')[0].src = this;
                        // Alternatively you could use:
                        // (new Image()).src = this;
                    });
                };
                this.rotateNext = function (b) {
                    (CONSOLE_MODE) ? console.log('bgSlider :: rotateNext :: called') : "";
                    if (!b) {
                        if ((this.imageList.length - 1) < this.nxtElem) {
                            this.nxtElem = 0;
                        }
                        var url = this.imageList[this.nxtElem];
                        //this.widgetElm.animate({opacity: 0}, 'fast');
                        //this.widgetElm.css('background-image', 'url(' + url + ')');
                        //this.widgetElm.animate({opacity: 1}, 'slow');
                        //
                        this.widgetElm.stop().animate({opacity: 0}, 1000, function () {
                            $(this).css({'background-image': "url('" + url + "')"})
                                .animate({opacity: 1}, {duration: 1000});
                        });
                        //
                        this.nxtElem++;
                    }
                    var that = this;//assign same object
                    this.rotateTick = setTimeout(function () {
                        that.rotateNext(false);
                    }, this.itemDuration);
                };
                this.initSlider = function () {
                    (CONSOLE_MODE) ? console.log('bgSlider :: initSlider :: called') : "";
                    this.widgetElm = $('#' + this.widgetID);
                    this.bgPreload();
                    this.rotateNext(true);
                };
                this.initSlider();
            }

            //End Added by Yesh

            // Dom Ready
            $(function () {
                parallax();
                // Step 02: call to function
                // Added by Yesh
                new bgSlider('myparallax');
                //End Added by Yesh
            });
        })(jQuery);

        /*
         jQuery Parallax 1.1.3
         Author: Ian Lunn
         Plugin URL: http://www.ianlunn.co.uk/plugins/jquery-parallax/

         Dual licensed under the MIT and GPL licenses:
         http://www.opensource.org/licenses/mit-license.php
         http://www.gnu.org/licenses/gpl.html
         */
        !function (n) {
            var t = n(window), e = t.height();
            t.resize(function () {
                e = t.height()
            }), n.fn.parallax = function (o, r, i) {
                function u() {
                    var i = t.scrollTop();
                    l.each(function (t, u) {
                        var l = n(u), f = l.offset().top, s = a(l);
                        i > f + s || f > i + e || l.css("backgroundPosition", o + " " + Math.round((l.data("firstTop") - i) * r) + "px")
                    })
                }

                var a, l = n(this);
                l.each(function (t, e) {
                    $element = n(e), $element.data("firstTop", $element.offset().top)
                }), a = i ? function (n) {
                    return n.outerHeight(!0)
                } : function (n) {
                    return n.height()
                }, (arguments.length < 1 || null === o) && (o = "50%"), (arguments.length < 2 || null === r) && (r = .1), (arguments.length < 3 || null === i) && (i = !0), t.bind("scroll", u).resize(u), u()
            }
        }(jQuery);
    </script>
    <?php
}

if (!(is_admin())) {
    wp_enqueue_script('jquery');
    add_action('wp_head', 'jquery_parallax');
}
```
Thanks!
